packages/amplify_datastore/ios/Classes/types/model/FlutterModelField.swift

```diff
@@ -20,65 +20,88 @@ import Amplify
     
     public let name: String
-    public let type: ModelFieldType
+    public let type: FlutterModelFieldType
     public let isRequired: Bool
     public let isArray: Bool
+    public let association: FlutterModelAssociation?
+    public let authRules: [FlutterAuthRule]?
+    
+    // Currently ignored
     public let attributes: [ModelFieldAttribute] = []
-    public let association: ModelAssociation? = nil
-    public let authRules: AuthRules = []
+
     
     init(serializedData: [String: Any]) {
         self.name = serializedData["name"] as! String
-        self.type = ModelFieldType.from(type: FlutterModelField.deserializeFieldType(type: serializedData["type"] as! String))
+        
+        self.type = FlutterModelFieldType(serializedData: serializedData["type"] as! [String: Any])
+        
         self.isRequired = serializedData["isRequired"] as! Bool
         self.isArray = serializedData["isArray"] as! Bool
+        
+        self.association = serializedData["association"] != nil ?  FlutterModelAssociation(serializedData: serializedData["association"] as! [String : Any]) :
+            nil
+        
+        self.authRules = (serializedData["authRules"] as? [[String:String]])?.map{
+            FlutterAuthRule(serializedData: $0)
+        }
+            
+            /*
+            (serializedData["authRules"] as? [[String:String]]).map({ (<#[[String : String]]#>) -> [String:String] in
+            FlutterAuthRule(serializedData: code)
+        })
+ */
+            //parseAuthRulesMap(authRulesMap: serializedData["authRules"] as? [[String:String]])
     }
     
-    static func deserializeFieldType(type: String) -> Any.Type {
-        if type == "String" {
-            return String.self
+    /*
+    private func parseAuthRulesMap(authRulesMap : [ [ String : String ]]?) -> [FlutterAuthRule]?{
+        if(authRulesMap != nil){
+            var result: [FlutterAuthRule] = [FlutterAuthRule]();
+            
+            for authRuleMap in authRulesMap! {
+                result.append( FlutterAuthRule(serializedData: authRuleMap));
+            }
+            
+            return result;
         }
-        if type == "Integer" {
-            return Int64.self
+        else{
+            return nil;
         }
-        if type == "Double" {
-            return Double.self
-        }
-        if type == "Boolean" {
-            return Bool.self
-        }
-        if type == "DateTime" {
-            return Temporal.DateTime.self
-        }
-        if type == "Time" {
-            return Temporal.Time.self
+    }
+ */
+    
+    private func convertFlutterAuthRules(flutterAuthRules : [FlutterAuthRule]?) -> [AuthRule]?{
+        if(flutterAuthRules != nil){
+            var result: [AuthRule] = [AuthRule]();
+            
+            for flutterAuthRule in flutterAuthRules! {
+                result.append( flutterAuthRule.convertToNativeModelField() );
+            }
+            
+            return result;
         }
```

Comment:
use if-let

Reply:
How do you imagine we'd use if-let in this context?  

flutterAuthRules is a input variable - are you suggesting we declare another variable?

Reply:
yeah basically you can do _I think_

```swift
if let flutterAuthRules = flutterAuthRules {
   ... // doesn't require ! anymore on flutterAuthRules since it's already confirmed to be non nil
}

Reply:
what would be the benefit of this approach?  This change seems to make the code less clear.

Reply:
This ensures that the variable is no longer _nil-able_ before you use it and cannot throw an unhandled exception.

Reply:
ah ok I made the change